deltacat/utils/metrics.py

```diff
@@ -0,0 +1,108 @@
+import ray
+import urllib
+import logging
+import yaml
+
+from deltacat import logs
+from enum import Enum
+from typing import Dict, Any, List, Callable
+from deltacat.aws.clients import resource_cache
+
+from botocore.exceptions import ClientError
+
+logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
+
+DEFAULT_DELTACAT_METRICS_NAMESPACE = "ray-deltacat-profiling"
+
+
+class MetricsType(str, Enum):
+    TIMER = "timer"
+
+
+class MetricsDimensionType(str, Enum):
+    NODE_IP = "node_ip"
+    RAY_TASK_ID = "task_id"
+    RAY_WORKER_ID = "worker_id"
+
+
+def _build_metrics_name(metrics_type: Enum, metrics_name: str) -> str:
+    metrics_name_with_type = f"{metrics_name}_{metrics_type}"
+    return metrics_name_with_type
+
+
+def _get_current_node_ip() -> str:
+    current_node_ip = urllib.request.urlopen(
+        "http://169.254.169.254/latest/meta-data/local-ipv4"
+    ).read().decode("utf-8")
+    return {"Name": f"node_ip",
+            "Value": f"{current_node_ip}"
+            }
+
+
+def _get_current_task_id() -> Dict[str, Any]:
+    return {"Name": f"ray_task_id",
+            "Value": f"{ray.get_runtime_context().get_task_id()}"
+            }
+
+
+def _get_current_worker_id() -> Dict[str, Any]:
+    return {"Name": f"ray_worker_id",
+            "Value": f"{ray.get_runtime_context().worker.core_worker.get_worker_id()}"
+            }
+
+
+def _get_current_instance_region() -> str:
+    current_instance_region = urllib.request.urlopen(
+        "http://169.254.169.254/latest/meta-data/placement/region"
+    ).read().decode("utf-8")
+    return current_instance_region
+
+
+def _get_current_cluster_name() -> str:
+    with open("/home/ubuntu/ray_bootstrap_config.yaml", "r") as f:
+        cluster_config = yaml.load(f)
+    return cluster_config["cluster_name"]
+
+
+METRICS_DIMENSION_TYPE_TO_VALUE_DICT: Dict[
```

Comment:
Minor: ideally use different modules for constants, if using the same module place constants above methods, and methods above classes.